src/main/java/com/nvs/store/service/CartService.java

```diff
@@ -0,0 +1,82 @@
+package com.nvs.store.service;
+
+
+import com.nvs.store.dto.cart.AddToCartDto;
+import com.nvs.store.dto.cart.CartDto;
+import com.nvs.store.dto.cart.CartItemDto;
+import com.nvs.store.exceptions.CustomException;
+import com.nvs.store.models.cart.Cart;
+import com.nvs.store.models.product.Product;
+import com.nvs.store.repository.CartRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+
+
+@Service
+public class CartService {
+    @Autowired
+    ProductService productService;
+
+    @Autowired
+    CartRepository cartRepository;
+
+    public void addToCart(AddToCartDto addToCartDto) {
+
+        Product product = productService.findById(addToCartDto.getProductId());
+       if (addToCartDto.getQuantity() <= product.getAvailable()) {
+           Cart cart = new Cart();
+           cart.setProduct(product);
+           cart.setQuantity(addToCartDto.getQuantity());
+           cart.setCreatedDate(new Date());
+
+           cartRepository.save(cart);
+       } throw new CustomException("We don't have enough available quantity of this product. We have only:" + product.getAvailable());
```

Comment:
The exception CustomException will be thrown only when there is no availability or all the time?